cmd/kubeadm/app/cmd/upgrade/upgrade.go

```diff
@@ -72,5 +75,7 @@ func addApplyPlanFlags(fs *pflag.FlagSet, flags *applyPlanFlags) {
 	fs.BoolVar(&flags.allowRCUpgrades, "allow-release-candidate-upgrades", flags.allowRCUpgrades, "Show release candidate versions of Kubernetes as an upgrade alternative and allow upgrading to a release candidate versions of Kubernetes.")
 	fs.BoolVar(&flags.printConfig, "print-config", flags.printConfig, "Specifies whether the configuration file that will be used in the upgrade should be printed or not.")
 	options.AddFeatureGatesStringFlag(fs, &flags.featureGatesString)
+	_ = fs.MarkDeprecated(options.FeatureGatesString, fmt.Sprintf("This flag is deprecated and will be removed in a future release. The upgrade workflow is not designed for reconfigure the cluster. "+
```

Comment:
if you agree with my prev comment we should also make this flag a no-op and mention that in the deprecation message.

we can remove it in the next release.
wuth the idea to not break user scripts right away.

Reply:
if you agree with the no-op part we should reflect it in the release note.

Reply:
Done